modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java

```diff
@@ -219,7 +219,11 @@ public Reader getTemplate(String name) {
             bundle.put("models", allModels);
             bundle.put("apiFolder", config.apiPackage().replace('.', File.separatorChar));
             bundle.put("modelPackage", config.modelPackage());
-            bundle.put("authMethods", config.fromSecurity(swagger.getSecurityDefinitions()));
+            List<CodegenSecurity> authMethods = config.fromSecurity(swagger.getSecurityDefinitions());
+            if (!authMethods.isEmpty()) {
```

Comment:
AuthMethods can be null, please fix it. 

if (authMethods != null && !authMethods.isEmpty()) {
                bundle.put("authMethods", authMethods);
                bundle.put("hasAuthMethods", true);
            }